Dockerfile

```diff
@@ -1,25 +1,71 @@
-FROM debian:stable
+ARG DEBIAN_VERSION=buster
+FROM debian:$DEBIAN_VERSION as builder
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    git-core \
+    cmake \
+    pkg-config \
+    libcurl4-openssl-dev \
+    libgnutls28-dev \
+    libsasl2-dev \
+    uuid-dev \
+    libtool \
+    libssl-dev \
+    libgcrypt20-dev \
+    libmicrohttpd-dev \
+    libltdl-dev \
+    libjson-c-dev \
+    libleptonica-dev \
+    libmosquitto-dev \
+    libunistring-dev \
+    dh-autoreconf \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /vzlogger
+
+RUN git clone https://github.com/volkszaehler/libsml.git --depth 1 \
+ && make install -C libsml/sml
+
+RUN git clone https://github.com/rscada/libmbus.git --depth 1 \
+ && cd libmbus \
+ && ./build.sh \
+ && make install
+
+COPY . /vzlogger
+
+RUN cmake -DBUILD_TEST=off \
+ && make \
+ && make install
+
+FROM debian:$DEBIAN_VERSION
 
 LABEL Description="vzlogger"
 
-# Dependencies 
-# https://wiki.volkszaehler.org/software/controller/vzlogger/installation_cpp-version
-RUN apt-get update && apt-get -y upgrade 
-RUN apt-get -y install build-essential git-core cmake pkg-config \
-    subversion libcurl4-openssl-dev libgnutls28-dev libsasl2-dev \
-    uuid-dev libtool libssl-dev libgcrypt20-dev libmicrohttpd-dev \
-    libltdl-dev libjson-c-dev libleptonica-dev libmosquitto-dev \
-    libunistring-dev dh-autoreconf sudo
-
-# Build from volkszaehler/vzlogger
-# pass "N" when prompted about adding systemd unit
-RUN mkdir /cfg && cd /tmp && \
-    git clone https://github.com/volkszaehler/vzlogger.git && \
-    cd vzlogger && \
-    echo "N\n" | bash ./install.sh
-
-# Setup volume 
+RUN apt-get update && apt-get install -y \
+    libcurl4 \
+    libgnutls30 \
+    libsasl2-2  \
+    libuuid1 \
+    libssl1.1 \
+    libgcrypt20  \
+    libmicrohttpd12 \
+    libltdl7 \
+    libatomic1 \
+    libjson-c3 \
+    liblept5 \
+    libmosquitto1 \
+    libunistring2 \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=builder /usr/local/bin/vzlogger /usr/local/bin/vzlogger
+COPY --from=builder /usr/local/lib/libmbus.so* /usr/local/lib/
```

Comment:
can we be sure that libsml is always linked statically?

Reply:
I have very limited knowledge of linking. I just took the steps from the install.sh. So long story short, to change the linking here I need help.

Reply:
no need to change "the linking",
i'm noting this because you copy libmbus to the target image, but not libsml.
most likely the build defaults to linking libsml statically.
ideally there should at least be a comment saying so, just in case that changes in the future.

Reply:
libsml compiles both a dynamic and static library.
this seems to at least prefer the static version:
https://github.com/volkszaehler/vzlogger/blob/master/modules/FindSML.cmake#L84
(how does that work? just by the ordering? i have no idea.)

Reply:
I added a comment